File: LeetCode/Best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minBPrice = INT_MAX;
        int profit = 0;
        for(int iter = 0,size = prices.size(); iter< size;iter++){
            if(prices[iter] < minBPrice){
                minBPrice = prices[iter];
            }
            if(profit  < (prices[iter] - minBPrice)){
                profit = prices[iter] - minBPrice;
            }
        }
        return profit;
    }
};

File: LeetCode/best_time_to_buy_and_sell_stock.java
class Solution {
    public int maxProfit(int[] prices) {
           if(prices.length == 0) return 0;
        int profit = 0;
        int minimum = prices[0];
        
        for(int iter =1 ; iter < prices.length; iter++){
            if(prices[iter] > minimum){
                if(profit < prices[iter] - minimum){
                    profit = prices[iter] - minimum;
                }
            }else if(prices[iter] < minimum){
                minimum = prices[iter];               
            }
        }
        return profit;
    }
}